app/controllers/concerns/mhv_controller_concerns.rb

```diff
@@ -14,24 +12,29 @@ module MHVControllerConcerns
   def authorize
     raise_access_denied if !current_user&.loa3? || current_user.mhv_account.ineligible?
     raise_requires_terms_acceptance if current_user.mhv_account.needs_terms_acceptance?
-    begin
-      current_user.mhv_account.create_and_upgrade! unless current_user.mhv_account.upgraded?
-    # TODO: rescue more specifically if mhv_account raises more specifically
-    ensure
-      raise_something_went_wrong unless current_user.mhv_account.upgraded?
-    end
+    create_mhv_account! unless current_user.mhv_account.upgraded?
```

Comment:
this could probably go into it's own before action, since it's not really related to authorization? :/